src/coreclr/vm/comutilnative.cpp

```diff
@@ -1774,8 +1775,21 @@ static ValueTypeHashCodeStrategy GetHashCodeStrategy(MethodTable* mt, QCall::Obj
                 }
                 else
                 {
-                    *fieldOffset += field->GetOffsetUnsafe();
-                    ret = GetHashCodeStrategy(fieldMT, objHandle, fieldOffset, fieldSize);
+                    MethodTable* valueTypeMT = CoreLibBinder::GetClass(CLASS__VALUE_TYPE);
+                    WORD slotGetHashCode = CoreLibBinder::GetMethod(METHOD__VALUE_TYPE__GET_HASH_CODE)->GetSlot();
+
+                    if (HasOverriddenMethod(mt, valueTypeMT, slotGetHashCode))
+                    {
+                        *fieldOffset += field->GetOffsetUnsafe();
+                        *method = mt->GetRestoredSlot(slotGetHashCode);
+                        ret = ValueTypeHashCodeStrategy::ValueTypeOverride;
+                    }
```

Comment:
I'm really not sure about the implementation here. Is it safe to pass PCODE as a managed function pointer?

The `GetHashCode` implementation of nested field may mutate the field while the outer struct is declared as `readonly`. The mutation would be visible for a boxed instance. Should the field be boxed instead? This will make it safer and easier to implement.

Reply:
This is complicated because of shared generics. You would also need to pass the hidden generic arg here.

I think it would be easier to just box in this case, similar to what `Equals` does.